Lib/wsgiref/headers.py

```diff
@@ -32,6 +34,11 @@ def __init__(self, headers=None):
         headers = headers if headers is not None else []
         if type(headers) is not list:
             raise TypeError("Headers must be a list of name/value tuples")
+        for header, value in headers:
+            if not _is_legal_header_name(header.encode('ascii')):
+                raise ValueError('Invalid header name %r' % (header,))
+            if _is_illegal_header_value(value.encode('ascii')):
+                raise ValueError('Invalid header value %r' % (value,))
```

Comment:
I would prefer to initialize self.headers to an empty list and then call "for header, value in headers: self.add_header(header, value)". Then remove self._convert_string_type(k) calls below.

Reply:
Added this. Why would we want to remove self._convert_string_type(k) calls though?